AlfrescoApp/View Controllers/Sync View Controller/RealmSyncViewController.m

```diff
@@ -0,0 +1,495 @@
+/*******************************************************************************
+ * Copyright (C) 2005-2016 Alfresco Software Limited.
+ *
+ * This file is part of the Alfresco Mobile iOS App.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+#import "RealmSyncViewController.h"
+#import "ConnectivityManager.h"
+#import "FileFolderCollectionViewCell.h"
+#import "SyncCollectionViewDataSource.h"
+#import "RealmSyncManager.h"
+#import "UniversalDevice.h"
+#import "FailedTransferDetailViewController.h"
+#import "PreferenceManager.h"
+#import "SyncNavigationViewController.h"
+
+static CGFloat const kCellHeight = 64.0f;
+static CGFloat const kSyncOnSiteRequestsCompletionTimeout = 5.0; // seconds
+
+static NSString * const kVersionSeriesValueKeyPath = @"properties.cmis:versionSeriesId.value";
+
+@interface RealmSyncViewController () < RepositoryCollectionViewDataSourceDelegate >
+
+@property (nonatomic) AlfrescoNode *parentNode;
+@property (nonatomic, strong) AlfrescoDocumentFolderService *documentFolderService;
+@property (nonatomic, strong) UIPopoverController *retrySyncPopover;
+@property (nonatomic, strong) AlfrescoNode *retrySyncNode;
+@property (nonatomic, assign) BOOL didSyncAfterSessionRefresh;
+
+@property (nonatomic, strong) UIBarButtonItem *switchLayoutBarButtonItem;
+
+@property (nonatomic, strong) SyncCollectionViewDataSource *dataSource;
+
+@end
+
+@implementation RealmSyncViewController
+
+- (id)initWithParentNode:(AlfrescoNode *)node andSession:(id<AlfrescoSession>)session
+{
+    self = [super initWithSession:session];
+    if (self)
+    {
+        self.session = session;
+        self.parentNode = node;
+        [[NSNotificationCenter defaultCenter] addObserver:self
```

Comment:
It would be nice to have all these notifications observers in a separate method. Could the observers from viewDidLoad be included too?